include/experimental/__p2630_bits/strided_index_range.hpp

```diff
@@ -0,0 +1,34 @@
+
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+namespace std {
+namespace experimental {
+
+// Slice Specifier allowing for strides and compile time extent
+template <class OffsetType, class ExtentType, class StrideType>
+struct strided_index_range {
+  using offset_type = OffsetType;
+  using extent_type = ExtentType;
+  using stride_type = StrideType;
+
+  OffsetType offset;
+  ExtentType extent;
+  StrideType stride;
+};
```

Comment:
Should we consider enforcing the Mandates here?

```suggestion
  static_assert(is_integral_v<OffsetType> || is_integral_constant_v<OffsetType>);
  static_assert(is_integral_v<ExtentType> || is_integral_constant_v<ExtentType>);
  static_assert(is_integral_v<StrideType> || is_integral_constant_v<StrideType>);
};
```

In the change above, `is_integral_constant_v` is defined as follows.

```c++
template<class T>
struct is_integral_constant : std::false_type {};

template<std::integral T, T Value>
struct is_integral_constant<std::integral_constant<T, Value>> : std::true_type {};

template<class T>
constexpr inline bool is_integral_constant_v = is_integral_constant<T>::value;
```